UI/src/app/modules/validation-result/services/filter.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject } from 'rxjs';
+import { FilterPayload } from '../components/filtering-form/filterPayload.interface';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class FilterService {
+
+  private filterState = new BehaviorSubject<FilterPayload>({
```

Comment:
please rename the file to validation-filter service, as we already have another file named filter.service that refer to dataset filtering